### PR TITLE
feat(engine): add escalation block builder

### LINKED ISSUE #10768: Add escalation events to the randomized property tests
## Description

We should add escalation events to the randomized process generator that is being used in the randomized property tests.

- Introduce a new `BlockBuilder` for escalation events
- This `BlockBuilder` should randomly decide if it will escalate from a subprocess or from a called activity
- The `BlockBuilder` should randomly decide if it will use an interrupting or a non-interrupting boundary event
- The `BlockBuilder` should randomly decide if it will throw using an intermediate throw event or an end event

The `BlockBuilder` will generate one of the following blocks:

**Subprocess**
Note the difference between exclusive and parallel gateways depending on the interruption of the boundary event.
image

**Call Activity**
The non-coloured parts show what the called process should look like. Note the difference between exclusive and parallel gateways depending on the interruption of the boundary event.
image

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Adds an Escalation event BlockBuilder for the BPMN randomized tests. The Escalation Event BlockBuilder generates a sub-process or a call activity. There is a child process that throws and escalation event, and there is a boundary escalation catch event (interupting or non-interrupting) that catches the escalation and joins in an appropriate gateway (exclusive or parallel).

The escalation code is unique to the generated block. At the moment, this might be redundant, as the escalation event will always be caught by the encapsulating boundary escalation catch event.

## Related issues

closes #10768

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
@koevskinikola Nice work 🚀 You missed the part of actually using this block 😄

I ran the test with this block a bunch of times and it failed a few times. I don't think it is related to your changes, so we might've found our first bug with it already. I'll retry it when you've made the changes.

File: test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/blocks/EscalationEventBuilder.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    return workInProgress
+        .moveToActivity(getElementId())
+        .boundaryEvent(boundaryEscalationCatchEventId)
+        .escalation(escalationCode)
+        .cancelActivity(usesInterruptingEscalationEvent)
+        .connectTo(joinGatewayId);
+  }
+
+  static class Factory implements BlockBuilderFactory {
```
[the remark this answers is not in the document]
Comment (koevskinikola) [reply test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/blocks/EscalationEventBuilder.java:148]: Oh no. I actually added it, commented out the rest of the factories for testing, and I guess I removed it when I reverted the Factory comments. :smile: 

However, I didn't get any failures. Maybe I didn't test correctly.

**Comment by koevskinikola** [comment]
@remcowesterhoud all done with implementing the review hints. Have a look when you can.

**Review by remcowesterhoud — APPROVED** [review]
LGTM 👍 

bors merge

**Pull request merged**